pkg/store/cache/caching_bucket.go

```diff
@@ -190,75 +466,80 @@ func (cb *CachingBucket) getRangeChunkFile(ctx context.Context, name string, off
 	}
 
 	// Start and end range are subrange-aligned offsets into object, that we're going to read.
-	startRange := (offset / cb.config.ChunkSubrangeSize) * cb.config.ChunkSubrangeSize
-	endRange := ((offset + length) / cb.config.ChunkSubrangeSize) * cb.config.ChunkSubrangeSize
-	if (offset+length)%cb.config.ChunkSubrangeSize > 0 {
-		endRange += cb.config.ChunkSubrangeSize
+	startRange := (offset / cfg.subrangeSize) * cfg.subrangeSize
+	endRange := ((offset + length) / cfg.subrangeSize) * cfg.subrangeSize
+	if (offset+length)%cfg.subrangeSize > 0 {
+		endRange += cfg.subrangeSize
 	}
 
 	// The very last subrange in the object may have length that is not divisible by subrange size.
-	lastSubrangeOffset := endRange - cb.config.ChunkSubrangeSize
-	lastSubrangeLength := int(cb.config.ChunkSubrangeSize)
+	lastSubrangeOffset := endRange - cfg.subrangeSize
+	lastSubrangeLength := int(cfg.subrangeSize)
 	if uint64(endRange) > size {
-		lastSubrangeOffset = (int64(size) / cb.config.ChunkSubrangeSize) * cb.config.ChunkSubrangeSize
+		lastSubrangeOffset = (int64(size) / cfg.subrangeSize) * cfg.subrangeSize
 		lastSubrangeLength = int(int64(size) - lastSubrangeOffset)
 	}
 
-	numSubranges := (endRange - startRange) / cb.config.ChunkSubrangeSize
+	numSubranges := (endRange - startRange) / cfg.subrangeSize
 
 	offsetKeys := make(map[int64]string, numSubranges)
 	keys := make([]string, 0, numSubranges)
 
-	for off := startRange; off < endRange; off += cb.config.ChunkSubrangeSize {
-		end := off + cb.config.ChunkSubrangeSize
+	totalRequestedBytes := int64(0)
+	for off := startRange; off < endRange; off += cfg.subrangeSize {
+		end := off + cfg.subrangeSize
 		if end > int64(size) {
 			end = int64(size)
 		}
+		totalRequestedBytes += (end - off)
 
 		k := cachingKeyObjectSubrange(name, off, end)
 		keys = append(keys, k)
 		offsetKeys[off] = k
 	}
 
 	// Try to get all subranges from the cache.
-	hits := cb.cache.Fetch(ctx, keys)
+	totalCachedBytes := int64(0)
+	hits := cache.Fetch(ctx, keys)
 	for _, b := range hits {
-		cb.fetchedChunkBytes.WithLabelValues(originCache).Add(float64(len(b)))
+		totalCachedBytes += int64(len(b))
+		cb.fetchedGetRangeBytes.WithLabelValues(originCache, cfgName).Add(float64(len(b)))
 	}
+	cb.operationHits.WithLabelValues(operationGetRange, cfgName).Add(float64(totalCachedBytes) / float64(totalRequestedBytes))
```

Comment:
Shouldn't we increase by `len(hits)`?

Reply:
I'd like for `operationHits` metric to be comparable to `operationRequests`, which is why I compute ratio of cached vs requested bytes. WDYT?

Reply:
Fixed by using `len(hits) / len(keys)`.